File: solutions/clojure/resistor-color-duo/1/src/resistor_color_duo.clj
(ns resistor-color-duo)


;; convert to Map
(def colors->num
  {"black" 0
   "brown" 1
   "red" 2
   "orange" 3
   "yellow" 4
   "green" 5
   "blue" 6
   "violet" 7
   "grey" 8
   "white" 9})

(defn resistor-value
  "Returns the resistor value based on the given colors"
  [colors]
  (->> colors
       (take 2)
       (map colors->num)
       (reduce (fn [acc x] (+ (* acc 10) x)))))

File: solutions/python/resistor-color-duo/1/resistor_color_duo.py
"""Resistor color duo helpers."""

COLORS = [
    "black",
    "brown",
    "red",
    "orange",
    "yellow",
    "green",
    "blue",
    "violet",
    "grey",
    "white",
]


def value(colors):
    """Return the resistor value represented by the first two colors."""
    first, second = colors[:2]
    return COLORS.index(first) * 10 + COLORS.index(second)
